pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -730,11 +730,11 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 
 	// Then give the plugin the chance to influence the decision on any
 	// remaining devices to allocate.
-	preferred, err := m.callGetPreferredAllocationIfAvailable(podUID, contName, resource, available.Union(devices), devices, required)
+	preferred, err := m.callGetPreferredAllocationIfAvailable(podUID, contName, resource, available.Union(allocated), allocated, required)
```

Comment:
question: read through the code a bit but im not completely clear on this diff.
Why use the union of `available` and `allocated` as the set of "available devices" and `allocated` for the set of "must include", instead of the union of the `available` and `devices` for all available and `devices` for the set that must be included?

Reply:
This was a bug that was't caught during review of the following PR (which landed yesterday):
https://github.com/kubernetes/kubernetes/pull/92665

It should always have been `available=available.Union(allocated)` and `mustInclude=allocated`, but only once I added the new tests in this PR was the bug uncovered.